enterprise/coderd/provisionerkeys.go

```diff
@@ -200,17 +200,42 @@ func (api *API) deleteProvisionerKey(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusNoContent, nil)
 }
 
+// @Summary Fetch provisioner key details
+// @ID fetch-provisioner-key-details
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Enterprise
+// @Param provisionerkey path string true "Provisioner Key"
+// @Success 200 {object} codersdk.ProvisionerKey
+// @Router /provisionerkeys/{provisionerkey} [get]
+func (*API) fetchProvisionerKey(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	pk, ok := httpmw.ProvisionerKeyAuthOptional(r)
```

Comment:
We should consider how this API will be used in the context of the old approach (i.e. pre-shared keys). This API endpoint will currently reject those requests with a `403 Forbidden`, which may not be the right approach. At the very least we should add a test case for this and perhaps add some special handling; `400 Bad Request` might be more appropriate, with a message indicating that a pre-shared key is not a provisioner key.

Reply:
Returning 400 for "I'm not letting you because you don't have the right secret" smells wrong to me. We can return 403 but include a message regarding the required auth header.

From the perspective of the provisioner daemon, it's going to check this endpoint if it has a provisioner key and needs to know what tagset is associated with the key. If we have a PSK we already _know_ what tags we have so don't need to check.

Reply:
Yeah, I can see your reasoning here. A 403 with a message will suffice 👍